options/options.go

```diff
@@ -178,7 +178,7 @@ func (terragruntOptions *TerragruntOptions) InsertTerraformCliArgs(argsToInsert
 
 	commandLength := 1
 	if util.ListContainsElement(TERRAFORM_COMMANDS_WITH_SUBCOMMAND, terragruntOptions.TerraformCliArgs[0]) {
-		commandLength = 2
+		commandLength = util.Min(2, cap(terragruntOptions.TerraformCliArgs))
```

Comment:
Use `len` (number of items in array), not `cap` (capacity of array).

Reply:
Also, I had to scratch my head at this logic for a min to understand it, so perhaps add a comment explaining why we're using `util.Min` here: e.g. "these Terraform commands require extra arguments, but it's possible the user didn't actually pass them, so use math.Min to make sure we don't try to read args that aren't there..."

Reply:
In this case cap and len will be the same number - I realized this after I sent the PR too. Only reason it's a cap because at first I had put it in the slices :D. I'll change to len as I think it makes more sense to use that now.

Will be adding docs and tests, shortly.